### PR TITLE
tcp chat added

### PR DESCRIPTION
**Pull request by Diminho**

### CODE REVIEW

File: CHAT_SERVER/chat_server.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+		name:     name[:len(name)-2], //triming \n chars
+		id:       identifier,
+		incoming: make(chan string),
+		writer:   writer,
+		reader:   reader}
+
+	return client
+}
+
+func (client *Client) listen() {
```
Comment (8tomat8) [inline CHAT_SERVER/chat_server.go]: Call `go client.listen()` will be more efficient than initialization of anon func in `client.listen()` =)

Comment (8tomat8) [reply CHAT_SERVER/chat_server.go]: Same in `room.startRoomChat()` =)

File: CHAT_SERVER/chat_server.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+const (
+	CONN_TYPE = "tcp"
+)
+
+type Client struct {
+	id       string
+	name     string
+	incoming chan string
+	reader   *bufio.Reader
```
Comment (8tomat8) [inline CHAT_SERVER/chat_server.go:22]: Why you use `bufio.Reder` and `bufio.Writer`? `net.Conn` that you got from `listener.Accept()` already implements both of them =)

Comment (Diminho) [reply CHAT_SERVER/chat_server.go:22]: Yes. You are right ;)

File: CHAT_SERVER/chat_server.go (hunk 1 above)
Comment (Diminho) [reply CHAT_SERVER/chat_server.go]: Maybe not efficient, but more readable...agree!

File: CHAT_SERVER/chat_server.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	name      string
+	connJoins chan net.Conn
+	incoming  chan string
+}
+
+func (room *Room) post(message string) {
+	for _, client := range room.clients {
+		index := strings.Index(message, "|")
+		if client.id != message[:index] {
+			client.writer.WriteString(message[index+1:])
```
Comment (arbrix) [inline CHAT_SERVER/chat_server.go]: these two instructions repeat several times, hm, what about oop?

File: CHAT_SERVER/chat_server.go [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+func main() {
+
+	room := createRoom("lobby")
+	listener, _ := net.Listen(CONN_TYPE, "localhost:8989")
+	defer listener.Close()
+	for {
+		connection, _ := listener.Accept()
+		room.connJoins <- connection
```
Comment (arbrix) [inline CHAT_SERVER/chat_server.go:300]: where connection close logic?

**Comment by arbrix** [comment]
Commands were not implemented

**Pull request closed without merging**